docs for webview2 installer preview (#749)
MicrosoftDocs/edge-developer: microsoft-edge/webview2/concepts/distribution.md

title: Distribution of Microsoft Edge WebView2 Application
author: MSEdgeTeam
ms.author: msedgedevrel
ms.date: 06/22/2020
ms.topic: conceptual
ms.prod: microsoft-edge
ms.technology: webview
keywords: IWebView2, IWebView2WebView, webview2, webview, wpf apps, wpf, edge, ICoreWebView2, ICoreWebView2Host, browser control, edge html
---

# Distribution of applications using WebView2  

> The WebView2 control utilizes the Microsoft Edge \(Chromium\) platform.  When packaging and distributing your app, make sure a copy of the platform or the WebView2 Runtime is present before the app starts.  The following page describes how you \(the developer\)are able to ensure that the WebView2 Runtime is installed and use of the two distribution modes for your WebView2 application:  [Evergreen](#evergreen-distribution-mode) and [Fixed version](#fixed-version-distribution-mode).  

Author: space after `(the developer)`

Reviewer: Updated.

## Evergreen distribution mode  

The evergreen distribution mode ensures that your app is taking advantage of the latest features and security updates.  The evergreen distribution mode has the following characteristics.  

*   The web platform updates automatically without additional effort from you.  
*   All applications that leverage the evergreen distribution mode use a shared copy of the platform binaries, which saves disk space.  

There are multiple channels WebView2 that applications may use as the web platform.  By default, WebView2 targets the most stable channel available on the device that meets the minimum version requirement of the WebView2 SDK.  The following channels are listed in order from most to least stable channel.  

1.  WebView2 Runtime \(Preview\)  
1.  Microsoft Edge Beta Channel  
1.  Microsoft Edge Dev Channel  
1.  Microsoft Edge Canary Channel    

> [!NOTE]
> The evergreen distribution model is recommended for most developers.  

> Microsoft Edge Stable Channel runtime is not a valid target for WebView2, and the reasons are described later.  

Author: Mark as IMPORTANT. Also, delete the word runtime after `Microsoft Edge Stable Channel`

Reviewer: Updated.
```md
> [!IMPORTANT]
> Microsoft Edge Stable Channel is not a valid target for WebView2, and the reasons are described later.  
```

> Do not ship WebView2 applications in production during the preview.  

The WebView team recommends the following workflow to ensure the Evergreen WebView2 Runtime is available.  

1.  Download the latest [Evergreen WebView2 Runtime Installer][Webview2Installer].  
1.  Include the installer in your application installer or updater.  
1.  During your application installation or update, check if the Evergreen WebView2 Runtime is already installed on the user machine.  If not, the application invokes the installer to install the runtime.  

Depending on your scenario, you may need to change the above workflow.  For example, your application installer may download the Evergreen WebView2 Runtime Installer instead of including it in your application package.  

> [!NOTE]
> Both the WebView2 Runtime and WebView2 Runtime installer are in preview.  The preview has a limited initial scope and is only available as a standalone, per-machine install on Windows 10 on x64.  In the future, support for Windows 7, x86, and ARM64 is planned.  

### Best practices for using WebView2 Runtime and non-stable Microsoft Edge channels  

Consider the following recommendations during the preview.  

> 1.  Make sure to use the [Evergreen WebView2 Runtime and Installer ][Webview2Installer] to develop or test your packaging and distribution pipeline.  In the future, your production application must include the installer.  

Author: `Must` is a strong word and not exactly true. I'd say is advised or should.

Reviewer: Updated.  
```md
1.  Make sure to use the [Evergreen WebView2 Runtime and Installer][Webview2Installer] to develop or test your packaging and distribution pipeline.  In the future, your production application should include the installer.  
```

1.  For developing your application, you may use the Evergreen WebView2 Runtime.  However, as the runtime shifts from the Dev Channel to the Beta Channel or Stable Channel, the runtime build number may not meet the most recent preview WebView2 SDK minimum version requirements.  If you wish to use the most recent SDK, install the Microsoft Edge Canary channel to ensure a compatible build is available on the device.  For more information about versioning, see [Versioning][ConceptsVersioning].  
1.  To test your web content for compatibility with changes to the platform not available in the Stable Channel, use the appropriate non-Stable Channel as required.  

## Fixed version distribution mode  

> [!NOTE]
> The fixed version distribution model is currently not available.  

For constrained environments, the WebView team plans to support a fixed version \(previously called bring-your-own\) distribution model.  The fixed version distribution model has the following benefits and limitations.  

> *   Allows you to select and package a specific version of the WebView2 Runtime in your application installer.  

Author: Can we use original wording in the doc? There are a lot of overlap in between bullets and this is not how I would describe benefits/limitations.

Reviewer: Updated.
```md
## Fixed version distribution mode  

> [!NOTE]
> The fixed version distribution model is currently not available.  

For constrained environments, there are plans to support a fixed version \(previously called bring-your-own\) distribution mode.  The fixed version distribution mode allows you to select and package a specific version of the WebView2 Runtime.  The fixed version distribution mode allows you to control which version of the WebView2 Runtime is used by your application, and when user machines are updated.  The fixed version distribution mode does not receive any automatic updates, and you should plan to manually apply updates.  
```

ms.topic: conceptual
ms.prod: microsoft-edge
ms.technology: webview
keywords: IWebView2, IWebView2WebView, webview2, webview, wpf apps, wpf, edge, ICoreWebView2, ICoreWebView2Host, browser control, edge html
---

# Distribution of applications using WebView2  

The WebView2 control utilizes the Microsoft Edge \(Chromium\) platform.  When packaging and distributing your app, make sure a copy of the platform or the WebView2 Runtime is present before the app starts.  The following page describes how you \(the developer\)are able to ensure that the WebView2 Runtime is installed and use of the two distribution modes for your WebView2 application:  [Evergreen](#evergreen-distribution-mode) and [Fixed version](#fixed-version-distribution-mode).  

## Evergreen distribution mode  

The evergreen distribution mode ensures that your app is taking advantage of the latest features and security updates.  The evergreen distribution mode has the following characteristics.  

*   The web platform updates automatically without additional effort from you.  
*   All applications that leverage the evergreen distribution mode use a shared copy of the platform binaries, which saves disk space.  

> You may choose from multiple channels WebView2 for applications to use as the web platform.  By default, WebView2 targets the most stable channel available on the device that meets the minimum version requirement of the WebView2 SDK.  The following channels are listed in order from most to least stable channel.  

Reviewer 2: ```suggestion
There are multiple channels WebView2 applications may use as the web platform.  By default, WebView2 targets the most stable channel available on the device that meets the minimum version requirement of the WebView2 SDK.  The following channels are listed in order from most to least stable channel.  
```

1.  WebView2 Runtime \(Preview\)  
1.  Microsoft Edge Beta Channel  
1.  Microsoft Edge Dev Channel  
1.  Microsoft Edge Canary Channel    

> [!NOTE]
> The evergreen distribution model is recommended for most developers.  

Microsoft Edge Stable Channel runtime is not a valid target for WebView2, and the reasons are described later.  

> For more information about versioning, see [Versioning][ConceptsVersioning].  
> 
> For more information about global methods, see [Globals][ReferenceWin3209538WebviewIdl].  

Reviewer 2: ```suggestion
For more information about versioning, see [Versioning][ConceptsVersioning] and [Globals][ReferenceWin3209538WebviewIdl].  
```

> ### Understand the WebView2 Runtime and installer  

Reviewer 2: ```suggestion
### Understand the WebView2 Runtime and installer (Preview)  
```

> > [!NOTE]
> > The following information supports the WebView2 Runtime \(Preview\).  

Reviewer 2: ```suggestion

```

Today during the preview, the Evergreen WebView2 Runtime and Microsoft Edge Dev Channel are updated at the same time and have the same build.  In the future during the Preview, the WebView2 Team plans to update the WebView2 Runtime and match the same build as the Microsoft Edge Beta Channel.  In the future when WebView2 reaches General Availability \(GA\), the WebView2 Team plans to update the WebView2 Runtime and match the same build as the Microsoft Edge Stable Channel.  After GA, applications should use the WebView2 Runtime in production.  

> [!IMPORTANT]
> Do not ship WebView2 applications in production during the preview.  

> The WebView team recommends the following workflow to ensure the Evergreen WebView2 Runtime is available.  

Reviewer 2: ```suggestion
We recommend the following workflow to ensure the Evergreen WebView2 Runtime is available.  
```

### Best practices for using WebView2 Runtime and non-stable Microsoft Edge channels  

Consider the following recommendations during the preview.  

1.  Make sure to use the [Evergreen WebView2 Runtime and Installer ][Webview2Installer] to develop or test your packaging and distribution pipeline.  In the future, your production application must include the installer.  
1.  For developing your application, you may use the Evergreen WebView2 Runtime.  However, as the runtime shifts from the Dev Channel to the Beta Channel or Stable Channel, the runtime build number may not meet the most recent preview WebView2 SDK minimum version requirements.  If you wish to use the most recent SDK, install the Microsoft Edge Canary channel to ensure a compatible build is available on the device.  For more information about versioning, see [Versioning][ConceptsVersioning].  
1.  To test your web content for compatibility with changes to the platform not available in the Stable Channel, use the appropriate non-Stable Channel as required.  

## Fixed version distribution mode  

> > [!NOTE]
> > The following information supports the fixed version distribution model.  The fixed version distribution model is currently not available.  

Reviewer 2: ```suggestion
> [!NOTE]
> The fixed version distribution model is currently not available.  
```
